src/main/java/hudson/plugins/release/pipeline/ReleaseStep.java

```diff
@@ -0,0 +1,218 @@
+    private static final Logger LOGGER = Logger.getLogger(ReleaseStep.class.getName());
+
+    private String job;
+    private List<ParameterValue> parameters;
+
+    @DataBoundConstructor
+    public ReleaseStep(String job) {
+        this.job = job;
+        parameters = new ArrayList<>();
+    }
+
+    public String getJob() {
+        return job;
+    }
+
+    public void setJob(String job) {
+        this.job = job;
+    }
+
+    public List<ParameterValue> getParameters() {
+        return parameters;
+    }
+
+    @DataBoundSetter public void setParameters(List<ParameterValue> parameters) {
+        this.parameters = parameters;
+    }
+
+    public static class Execution extends AbstractStepExecutionImpl {
+        @StepContextParameter private transient Run<?,?> invokingRun;
+        @StepContextParameter private transient TaskListener listener;
+
+        @Inject(optional=true) transient ReleaseStep step;
+
+        private List<ParameterValue> updateParametersWithDefaults(AbstractProject project,
+                List<ParameterValue> parameters) throws AbortException {
+
+            if (project instanceof BuildableItemWithBuildWrappers) {
+                ReleaseWrapper wrapper = ((BuildableItemWithBuildWrappers) project).getBuildWrappersList()
+                                                                                   .get(ReleaseWrapper.class);
+                if (wrapper != null) {
+                    for (ParameterDefinition pd : wrapper.getParameterDefinitions()) {
+                        boolean parameterExists = false;
+                        for (ParameterValue pv : parameters) {
+                            if (pv.getName()
+                                  .equals(pd.getName())) {
+                                parameterExists = true;
+                                break;
+                            }
+                        }
+                        if (!parameterExists) {
+                            parameters.add(pd.getDefaultParameterValue());
+                        }
+                    }
+                } else {
+                    throw new AbortException("Job doesn't have release plugin configuration");
+                }
+            }
+            return parameters;
+        }
+
+        @Override
+        public boolean start() throws Exception {
+            if (step.getJob() == null) {
+                throw new AbortException("Job name is not defined.");
+            }
+
+            final AbstractProject project = Jenkins.getActiveInstance()
+                                                   .getItem(step.getJob(), invokingRun.getParent(), AbstractProject.class);
+            if (project == null) {
+                throw new AbortException("No parametrized job named " + step.getJob() + " found");
+            }
+            listener.getLogger().println("Releasing project: " + ModelHyperlinkNote.encodeTo(project));
+
+            List<Action> actions = new ArrayList<>();
```

Comment:
NIT: reserve space for 3 items since you already know how much items will be put